src/RestClient/Authenticators/OAuth/src/OAuthAuthenticator.cs

```diff
@@ -58,6 +60,14 @@ public virtual async Task AuthenticateAsync(IClientRequest request, Cancellation
             request.AddHeader("Authorization", "Bearer " + oAuthToken);
         }
 
+        public virtual async Task RevokeTokensAsync()
+        {
+            foreach (var ts in _tokenSources)
```

Comment:
I think revoking the token should also remove it from the stores since its now revoked, theres no reason for it to still exist in there.

Im also not sure if it makes sense for this to be on the `IAuthenticator` interface as its only really relevant to the `OAuthAuthenticator` (since thats the one that has all the stores and stuff in it)

Reply:
Yeah, it is not an ideal place for this. Remove it from `IAuthenticator` for now and will revisit this in the next sprint.